contracts/EverlongStrategy.sol

```diff
@@ -520,6 +546,183 @@ contract EverlongStrategy is BaseStrategy {
+        _wrappedAmount = IERC20Wrappable(address(asset)).wrap(_unwrappedAmount);
+    }
+
+    /// @dev Unwrap the strategy asset so that it can be used with hyperdrive.
+    /// @param _wrappedAmount Amount of wrapped strategy assets to unwrap.
+    /// @return _unwrappedAmount Amount of unwrapped tokens received.
+    function _unwrap(
+        uint256 _wrappedAmount
+    ) internal returns (uint256 _unwrappedAmount) {
+        // If the strategy doesn't use a wrapped asset, revert.
+        if (!isWrapped) {
+            revert IEverlongStrategy.AssetNotWrapped();
+        }
+
+        // Unwrap the strategy assets.
+        _unwrappedAmount = IERC20Wrappable(address(asset)).unwrap(
+            _wrappedAmount
+        );
+    }
+
+    /// @dev Calculate the maximum amount of strategy assets that can be spent
+    ///      on longs before hyperdrive runs out of liquidity.
+    /// @return assets Max amount of strategy assets that can be spent on longs.
+    function _calculateMaxLong() internal view returns (uint256 assets) {
+        // Retrieve the max long amount denominated in hyperdrive base tokens.
+        assets = IHyperdrive(hyperdrive).calculateMaxLong();
+
+        // If `asBase` is false, convert the amount to be denominated in vault
+        // share tokens.
+        if (!asBase) {
+            assets = IHyperdrive(hyperdrive)._convertToShares(assets);
+        }
+    }
+
+    /// @dev Open a long with the specified amount of assets. Return the amount
+    ///      of bonds received and their maturityTime.
+    /// @param _toSpend Amount of strategy assets to spend.
+    /// @param _minOutput Minimum amount of bonds to accept.
+    /// @param _minVaultSharePrice Minimum hyperdrive vault share price to
+    ///                            purchase at.
+    /// @param _extraData Extra data to pass to hyperdrive.
+    /// @return maturityTime Maturity time for bonds received.
+    /// @return bondAmount Amount of bonds received.
+    function _openLong(
+        uint256 _toSpend,
+        uint256 _minOutput,
+        uint256 _minVaultSharePrice,
+        bytes memory _extraData
+    ) internal returns (uint256 maturityTime, uint256 bondAmount) {
+        // Prepare for opening the long differently if the strategy asset is
+        // a wrapped hyperdrive token.
+        if (isWrapped) {
+            // The strategy asset is a wrapped hyperdrive token so it must be
+            // unwrapped.
+            _toSpend = _unwrap(_toSpend);
+
+            // Approve hyperdrive for the unwrapped asset, which is also the
+            // `hyperdriveToken`.
+            //
+            // Leave the approval slot dirty to save gas on future approvals.
+            ERC20(hyperdriveToken).forceApprove(
+                address(hyperdrive),
+                _toSpend + 1
+            );
+
+            // Convert back to hyperdrive's base denomination, same as the
+            // wrapped token's.
+            _toSpend = convertToWrapped(_toSpend);
```

Comment:
The wrapped token is denominated in units of vault shares, not base tokens.

Reply:
Agree that the comment is confusing, but I think the logic is correct...

Correct, however the `_toSpend = _unwrap(_toSpend);` call above essentially calls `convertToBase` on the `_toSpend` which is needed for the approval, but must be reverted by calling `convertToWrapped` (aka `convertToShares`) . Doing this also comes with the benefit of conservative rounding to avoid off-by-ones

Reply:
Yep I agree. I was commenting on the comment rather than the logic. This is correct.